src/ecdsa/ecdh.py

```diff
@@ -38,18 +38,25 @@ class InvalidSharedSecretError(Exception):
 class ECDH(object):
     """
     Elliptic-curve Diffie-Hellman (ECDH). A key agreement protocol.
+    
     Allows two parties, each having an elliptic-curve public-private key 
     pair, to establish a shared secret over an insecure channel
     """""
 
     def __init__(self, curve=None, private_key=None, public_key=None):
         """
-        ECDH init. Class can be init without parameters and then
-        it will needs to set all of them for proper operation.
+        ECDH init.
+
+        Class can be init without parameters and then it will needs to set
```

Comment:
"Call can be initialised without parameters, then the first operation (loading either the key) will set the used curve.
All parameters must be ultimately set before share secret calculation will be allowed." ?